lib/routes.js

```diff
@@ -213,6 +215,37 @@ function verifyTx (req, res, next) {
     .catch(next)
 }
 
+function verifyCoupon (req, res, next) {
+  couponManager.getCoupon(req.body.codeInput)
+    .then(coupon => {
+      if (!coupon) return next()
+
+      const transaction = req.body.tx
+      const commissions = configManager.getCommissions(transaction.cryptoCode, req.deviceId, req.settings.config)
+      const tickerRate = BN(transaction.rawTickerPrice)
+      const discount = getDiscountRate(coupon.discount, commissions[transaction.direction])
+      const rates = {
+        [transaction.cryptoCode]: {
+          [transaction.direction]: (transaction.direction === 'cashIn')
+            ? tickerRate.mul(discount).round(5)
+            : tickerRate.div(discount).round(5)
+        }
+      }
+
+      respond(req, res, {
+        coupon: coupon,
+        newRates: rates
+      })
+    })
+    .catch(next)
+}
+
+function getDiscountRate (discount, commission) {
+  const percentageDiscount = BN(1).sub(BN(discount).div(100))
+  const percentageCommission = BN(commission).div(100)
+  return BN(1).add(percentageDiscount.mul(percentageCommission))
+}
```

Comment:
Isn't this the same function that was added into plugins but without the defaulting to zero? Might as well just call the other function, create a `commission-math.js` or similar and add the calc functions there.